JavaWeaver/api/kadabra/analysis/energy/detectors/ExcessiveMethodCallsDetector.js

```diff
@@ -1,21 +1,23 @@
 laraImport("kadabra.analysis.energy.detectors.BaseDetector");
 
-const containsJP = (arr, jp) => arr.some((i) => i.same(jp));
+class ExcessiveMethodCallsDetector extends BaseDetector {
+  constructor(debugEnabled = false) {
+    super("Excessive Method Calls Detector", debugEnabled);
+  }
 
-const tryGetMethodInfo = (arr, jp) => {
-  for (let j of arr) {
-    if (j.method.same(jp)) return j;
+  static containsJP(arr, jp) {
+    return arr.some((i) => i.same(jp));
   }
-  return null;
-};
 
-const addIfNew = (arr, jp) => {
-  if (!containsJP(arr, jp)) arr.push(jp);
-};
+  static tryGetMethodInfo(arr, jp) {
+    for (let j of arr) {
+      if (j.method.same(jp)) return j;
+    }
+    return null;
+  }
 
-class ExcessiveMethodCallsDetector extends BaseDetector {
-  constructor(debugEnabled = false) {
-    super("Excessive Method Calls Detector", debugEnabled);
+  static addIfNew(arr, jp) {
+    if (!this.containsJP(arr, jp)) arr.push(jp);
```

Comment:
Since `containsJP` is a static method, it is clearer if it is written as `ExcessiveMethodCallsDetector.containsJP` instead of `this.containsJP`. Check similar occurences.

If the method will be used several times inside a function, you could also do something like:

```
const containsJP = ExcessiveMethodCallsDetector.containsJP;
// Use containsJP as before, as a global function
```